### PR TITLE
feat: Add basic package infos and resource links to sidebar.

### PR DESCRIPTION
**Pull request by pkaske**
It's my very first time working with this kind of front end stack. From webpack to react, so I hope I didn't write too terrible code.

I've added a new module to the sidebar that shows a link to the package homepage, repository and license if available in the meta data.

It's a very basic implementation that probably doesn't catch a lot of the edge cases (especially in regards of the repository url), but maybe a starting point?

auswahl_001

**Type:**

The following has been addressed in the PR:

*  There is a related issue?
*  Unit or Functional tests are included in the PR

**Description:**

### CODE REVIEW

File: test/webui/components/PackageSidebar/infos.spec.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/**
+ * LastSync component
```
Comment (ayusharma) [inline test/webui/components/PackageSidebar/infos.spec.js]: Change it to Info Component.

File: src/webui/src/components/PackageSidebar/modules/Infos/index.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  get infos() {
+    const homepage = this.normalizeInfo(get(this, 'props.packageMeta.latest.homepage', null));
+    const repo = this.normalizeInfo(get(this, 'props.packageMeta.latest.repository', null));
+    const license = get(this, 'props.packageMeta.latest.license', 'N/A');
+
+    return {homepage, repo, license};
+  }
+
+  normalizeInfo(infoObj) {
+    if (typeof infoObj === 'string') {
```
Comment (juanpicado) [inline src/webui/src/components/PackageSidebar/modules/Infos/index.jsx]: Use here `lodash` `isString`. Mostly to have consistency along the project.

```
import isString from 'lodash/isString';
```

File: src/webui/src/components/PackageSidebar/modules/Infos/index.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      return {url: infoObj};
+    } else if (infoObj === null) {
+      return {url: ''};
+    }
+
+    infoObj.url = infoObj.url.replace(/^git\+/, '');
```
Comment (juanpicado) [inline src/webui/src/components/PackageSidebar/modules/Infos/index.jsx]: Move this to utilities (you must create a function there)  for easy testing.

```
infoObj.url = normalizeGitUrl(infoObj.url).
```

File: src/webui/src/components/PackageSidebar/modules/Infos/index.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+    return infoObj;
+  }
+
+  render() {
+    const infos = this.infos;
+
+    if (!infos.homepage.url && !infos.repo.url && infos.license === 'N/A') {
```
Comment (juanpicado) [inline src/webui/src/components/PackageSidebar/modules/Infos/index.jsx]: To avoid weird behavior, let's use `isNil` here.

`import isNil from 'lodash/isNil';`

File: src/webui/src/components/PackageSidebar/modules/Infos/index.jsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        title="Infos"
+        className={classes.infosModule}
+      >
+        <ul>
+          {infos.homepage.url &&
+            <li><span>Homepage</span><a href={infos.homepage.url} target="_blank">{infos.homepage.url}</a></li>
+          }
+
+          {infos.repo.url &&
+            <li><span>Repository</span><a href={infos.repo.url} target="_blank">{infos.repo.url}</a></li>
```
Comment (juanpicado) [inline src/webui/src/components/PackageSidebar/modules/Infos/index.jsx]: `<li><span>Homepage</span><a href={infos.homepage.url} target="_blank">{infos.homepage.url}</a></li>`

That line is duplicated, I guess you can create a `renderSection()` function in this class and reuse it.

File: src/webui/src/components/PackageSidebar/modules/Infos/style.scss [hunk 6]
```diff
@@ -0,0 +1,22 @@
+@import '../../../../styles/variable';
+
+.infosModule {
+  li {
+    display: flex;
+    font-size: 14px;
```
Comment (juanpicado) [inline src/webui/src/components/PackageSidebar/modules/Infos/style.scss]: We don't have a font size variable here, but It is a good moment to create one instead hardcode font in this file. 
https://github.com/verdaccio/verdaccio/blob/master/src/webui/src/styles/variable.scss#L21

File: src/webui/src/components/PackageSidebar/modules/Infos/index.jsx (hunk 3 above)
Comment (pkaske) [reply src/webui/src/components/PackageSidebar/modules/Infos/index.jsx]: Does "move this to utilities" refer to some other file? For now I just added a function to the component file itself.

File: src/webui/src/components/PackageSidebar/modules/Infos/style.scss (hunk 6 above)
Comment (pkaske) [reply src/webui/src/components/PackageSidebar/modules/Infos/style.scss]: All of the other modules in the sidebar seem to hard code the font size currently. Would it maybe better to create a separate pr and change all of them at once?

File: src/webui/src/components/PackageSidebar/modules/Infos/index.jsx (hunk 4 above)
Comment (pkaske) [reply src/webui/src/components/PackageSidebar/modules/Infos/index.jsx]: The url property actually is an empty string instead of null if not provided. I changed the code to `...url === ''`.
Is that ok?

File: src/webui/src/components/PackageSidebar/modules/Infos/index.jsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+  normalizeGitUrl(infoObj) {
+    return infoObj.url.replace(/^git\+/, '');
+  }
+
+  render() {
+    const infos = this.infos;
+
+    if (infos.homepage.url === '' && infos.repo.url === '' && infos.license === 'N/A') {
+      return '';
```
Comment (juanpicado) [inline src/webui/src/components/PackageSidebar/modules/Infos/index.jsx]: Just wondering, what happens whether the repository has no license but has homepage and url? Do you want to do not display at least what is available? What do you think?

Comment (juanpicado) [reply src/webui/src/components/PackageSidebar/modules/Infos/index.jsx]: Nevermind ! I did not see conditions bellow.

File: src/webui/src/components/PackageSidebar/modules/Infos/style.scss (hunk 6 above)
Comment (juanpicado) [reply src/webui/src/components/PackageSidebar/modules/Infos/style.scss]: Ok, make sense. I'll add a note in the projects section.

File: test/webui/components/PackageSidebar/infos.spec.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+import Infos from '../../../../src/webui/src/components/PackageSidebar/modules/Infos';
+import { packageMeta } from '../store/packageMeta';
+ 
+console.error = jest.fn();
+
+describe('<PackageSidebar /> : <Infos />', () => {
+  it('should load the component and check getter: info with package data', () => {
+    const wrapper = mount(<Infos packageMeta={packageMeta} />);
+    const instance = wrapper.instance();
+    const result = {
```
Comment (juanpicado) [inline test/webui/components/PackageSidebar/infos.spec.js]: Just being a bit picky here. 
https://codecov.io/gh/verdaccio/verdaccio/pull/757/diff
Perhaps we should add different scenarios here as missing homepage, repo or license. But in a new PR if you wish.

**Comment by juanpicado** [comment]
Thanks @pkaske great and useful addition !! It will ready in the next minor release soon.

File: test/webui/components/PackageSidebar/infos.spec.js (hunk 8 above)
Comment (pkaske) [reply test/webui/components/PackageSidebar/infos.spec.js]: That's probably a good idea. How should I go about serving different version of meta data? Deep copy `packageMeta` and modify it would be one way I guess. If that's ok, is there a copy function in the test framework available or should I utilize lodash?
